metafix/src/main/java/org/metafacture/metafix/FixPath.java

```diff
@@ -276,7 +279,15 @@ else if (isReference(field)) {
         }
         else {
             if (!hash.containsField(field)) {
-                hash.put(field, Value.newHash().withPathSet(newValue.getPath()));
+                final Value value = Arrays.asList(ReservedField.$prepend.name(), ReservedField.$append.name())
+                        .contains(tail(path)[0]) ? Value.newArray() : Value.newHash();
+                hash.put(field, value.withPathSet(newValue.getPath()));
+            }
+            else {
+                final Value value = hash.get(field);
+                if (value.isString()) {
+                    hash.put(field, Value.newArray(a -> a.add(value)));
```

Comment:
Shouldn't this only apply to `$append`/`$prepend`?

Reply:
Added conditional and updated test to document the behavior: https://github.com/metafacture/metafacture-fix/pull/368/commits/c5aa947a99a8e27a3a6b5c7ac3b75a1fc42d001e